Wangscape/tilegen/TilesetGenerator.cpp

```diff
@@ -30,21 +40,31 @@ void TilesetGenerator::generate(std::function<void(const sf::Texture&, std::stri
     {
         CornersGenerator corners_generator(clique, static_cast<size_t>(CORNERS));
         std::pair<size_t, size_t> tileset_resolution = corners_generator.size2D(options.tileFormat.resolution);
-        size_t res_x = tileset_resolution.first;
-        size_t res_y = tileset_resolution.second;
-        std::unique_ptr<sf::RenderTexture> output{getBlankImage(res_x, res_y)};
+        mResX = tileset_resolution.first;
+        mResY = tileset_resolution.second;
+        std::unique_ptr<sf::RenderTexture> output{getBlankImage(mResX, mResY)};
+        mDebugTilesets.clear();
 
-        auto filename = getOutputImageFilename(clique);
+        auto filename_base = getOutputImageFilename(clique);
+        auto filename = filename_base + options.tileFormat.fileType;
 
         // MetaOutput.addTileset, addTile should use this version of filename;
         // relative to output dir, not options dir!
-        metaOutput.addTileset(clique, filename, res_x, res_y);
+        metaOutput.addTileset(clique, filename, mResX, mResY);
         generateClique(clique, *output, filename);
         output->display();
 
         p.append(filename);
         callback(output->getTexture(), p.string());
         p.remove_filename();
+        for (auto& it : mDebugTilesets)
```

Comment:
I don't think that would work, because the non-`const` method `display()` is called on `*it.second`.